### DISCUSSION TITLE
using pycoilgen for supercon magnet design

### DISCUSSION THREAD

**Discussion by leeoral** [kev-m/pyCoilGen#73] [Q&A] [answered]
Hi there. Do you know if it is possible to use pycoilgen for magnet design, namely, generating a homogeneous field in one direction given a single or multilayer surface mesh? Also, are you aware of anyone who used the simulation results to design supercon magnets?

Thanks!

**Comment by kev-m** [comment, marked as answer]
Hello! pyCoilGen is a generic tool for calculating a wire path that satisfies provided solution boundaries, where the boundaries are the wire path surface and the desired magnetic in a volume. 

I understand that a homogenous field is one with no gradient, i.e. specify that your gradient is 0?

If you want multiple layers, where each layer gets a single wire path, you will have to specify this as a multi-layered STL file. Each non-connected layer will be identified and used.

I have no idea about using superconducting tape to wind the final wirepath.

**Reply by leeoral** [reply]
Thanks for the reply. Do you have any examples with homogeneous field and zero gradient?

**Reply by kev-m** [reply]
Hello, unfortunately the examples are all constructed to support MRI coils, which have a gradient field by design. 

I suggest that you try one of the trivial examples, e.g. [with a planar wire mesh, set the field function to a constant](https://github.com/kev-m/pyCoilGen/blob/master/examples/biplanar_xgradient.py#L18), and examine the generated magnetic field, similar to the examples [in the documentation](https://pycoilgen.readthedocs.io/en/stable/results.html#the-target-field-z-component).

**Reply by kev-m** [reply]
Otherwise, you can [manually specify a magnetic field (x,y,z,b)](https://pycoilgen.readthedocs.io/en/stable/configuration.html#using-a-numpy-pickle-file) and provide it to the program.

**Reply by leeoral** [reply]
This is wonderful. Thank you for the input!
